api/src/main/java/net/md_5/bungee/api/QueryResponse.java

```diff
@@ -0,0 +1,48 @@
+package net.md_5.bungee.api;
+
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.Setter;
+import lombok.ToString;
+import net.md_5.bungee.api.config.ListenerInfo;
+import net.md_5.bungee.api.connection.ProxiedPlayer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Data
+@ToString
+public class QueryResponse
+{
+
+    // Basic & full query
+    private String motd;
+    private int playerCount;
+    private int maxPlayers;
+    private int port;
+    private String address;
+
+    // full query only
+    private String version;
+    private String world = "BungeeCord_Proxy";
+    private String server = "BungeeCord";
+    @Setter(AccessLevel.NONE)
+    private List<String> plugins = new ArrayList<>();
+    @Setter(AccessLevel.NONE)
+    private List<String> players = new ArrayList<>();
+
+    public QueryResponse( ProxyServer server, ListenerInfo listener )
+    {
+        motd = listener.getMotd();
+        playerCount = server.getOnlineCount();
+        maxPlayers = listener.getMaxPlayers();
+        port = listener.getHost().getPort();
+        address = listener.getHost().getHostString();
+        version = server.getGameVersion();
+
+        for ( ProxiedPlayer player : server.getPlayers() )
```

Comment:
Maybe a way to hide players? Some servers like to do this.

Reply:
Query can be disable in the config, and it s already like this in BungeeCord and is not really related to the ProxyQueryEvent

Reply:
I'd argue that whilst it may mirror existing behaviour: it would be an ideal addition for this PR should plugins use this event and want that functionality.

Reply:
A plugin can clear/remove elements in the players list

Reply:
Ah my bad, I saw the setter being disabled and immediately jumped to the conclusion that you didn't want plugins to change anything in it. Looks good to me then.